### PR TITLE
Add more information for natives

### PR DESCRIPTION
**Pull request by AvarianKnight**
Some of the natives lack good information on what they do and examples on how to use them, this is an attempt to improve it somewhat.

This is still a WIP PR and will be updated periodically, there will be certain natives that have `## VALIDATE ME` that need validation on their actual usage.

If anyone sees anything that is wrong please make sure to leave a review

This PR will be all rebased into one commit when its no longer a WIP, but reviewing is likely easier across multiple commits

### CODE REVIEW

**Comment by AvarianKnight** [comment]
Do natives that have their preceding underscore removed need to be aliased?

**Comment by thorium-cfx** [comment]
> Do natives that have their preceding underscore removed need to be aliased?

Seems like it, these aliases are used to (at least) add hashes/items to the `Hash` enum in C#. Not having the _ prefixed version would then break backwards compatibility.

File: ENTITY/GetEntityCanBeDamaged.md [hunk 1]
```diff
@@ -9,8 +9,10 @@ aliases: ["0xD95CC5D2AB15A09F"]
 BOOL GET_ENTITY_CAN_BE_DAMAGED(Entity entity);
 ```
 
+Getter for [`SET_ENTITY_CAN_BE_DAMAGED`](_0x1760FFA8AB074D66)
```
Comment (4mmonium) [inline ENTITY/GetEntityCanBeDamaged.md:12]: Slight nitpick, this is supposed to have a preceding `#` before the underscore `_` i.e. `#_0x1760FFA8AB074D66`. Noticed this multiple times within this PR, missed it a couple times.  Not adding the hash will cause a 404 when clicked.

Comment (AvarianKnight) [reply ENTITY/GetEntityCanBeDamaged.md:12]: I do believe you are correct.

File: ENTITY/GetEntityType.md [hunk 2]
```diff
@@ -8,12 +8,13 @@ ns: ENTITY
 int GET_ENTITY_TYPE(Entity entity);
 ```
 
-```
-Returns:
-0 = no entity
-1 = ped
-2 = vehicle
-3 = object
+```c
```
[the remark this answers is not in the document]
Comment (AvarianKnight) [reply ENTITY/GetEntityType.md:11]: This is the method used in [other natives](https://github.com/citizenfx/fivem/edit/master/ext/native-decls/GetRopeFlags.md)

File: ENTITY/HasEntityClearLosToEntityAdjustForCover.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+ns: ENTITY
+aliases: ["0x394BDE2A7BBA031E", "_HAS_ENTITY_CLEAR_LOS_TO_ENTITY_2"]
+---
+## HAS_ENTITY_CLEAR_LOS_TO_ENTITY_ADJUST_FOR_COVER
+
+```c
+// 0x394BDE2A7BBA031E
+cs_type(Any) BOOL HAS_ENTITY_CLEAR_LOS_TO_ENTITY_ADJUST_FOR_COVER(Entity entity1, Entity entity2, int traceType);
+```
+
```
Comment (4mmonium) [inline ENTITY/HasEntityClearLosToEntityAdjustForCover.md:11]: The NativeDB introduced comments are somewhat useful, they're automatically added by the importer for natives and they allow you to know in which game build a native was added. I would suggest preserving them.

```
NativeDB Introduced: v1868
```

File: ENTITY/SetEntityProofs.md [hunk 4]
```diff
@@ -8,11 +8,9 @@ ns: ENTITY
...
 ```
 
-```
-Enable / disable each type of damage.  
---------------  
-p7 is to to '1' in am_mp_property_ext/int: entity::set_entity_proofs(uParam0->f_19, true, true, true, true, true, true, 1, true);  
-```
+Makes the specified entity invulnerable to the specified damage type
+
+Setter for [`GET_ENTITY_PROOFS`](_0xBE8CD9BE829BBEBF)
```
Comment (4mmonium) [inline ENTITY/SetEntityProofs.md:13]: Needs a preceding `#`.

Setter for [`GET_ENTITY_PROOFS`](#_0xBE8CD9BE829BBEBF)

File: ENTITY/SetObjectAsNoLongerNeeded.md [hunk 5]
```diff
@@ -8,9 +8,7 @@ ns: ENTITY
 void SET_OBJECT_AS_NO_LONGER_NEEDED(Object* object);
 ```
 
-```
-This is an alias of SET_ENTITY_AS_NO_LONGER_NEEDED.  
-```
+Alias for [`SET_ENTITY_AS_NO_LONGER_NEEDED`](_0xB736A491E64A32CF)
```
Comment (4mmonium) [inline ENTITY/SetObjectAsNoLongerNeeded.md:11]: Needs preceding `#`.

File: ENTITY/SetPedAsNoLongerNeeded.md [hunk 6]
```diff
@@ -8,9 +8,7 @@ ns: ENTITY
 void SET_PED_AS_NO_LONGER_NEEDED(Ped* ped);
 ```
 
-```
-This is an alias of SET_ENTITY_AS_NO_LONGER_NEEDED.  
-```
+Alias for [`SET_ENTITY_AS_NO_LONGER_NEEDED`](_0xB736A491E64A32CF)
```
Comment (4mmonium) [inline ENTITY/SetPedAsNoLongerNeeded.md:11]: Needs preceding `#`.

**Review by 4mmonium — CHANGES_REQUESTED** [review]
Reviewed most of this, some changes needed, mainly for stuff like this: ```Alias for [`SET_ENTITY_AS_NO_LONGER_NEEDED`](_0xB736A491E64A32CF)``` where preceding hashes are needed. Other than that looks good to me for the most part 😊

There are conflicting files (which is normal since a while has passed), but we can work through that later I guess.

File: ENTITY/GetEntityType.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (4mmonium) [reply ENTITY/GetEntityType.md:11]: > This is the method used in [other natives](https://github.com/citizenfx/fivem/edit/master/ext/native-decls/GetRopeFlags.md)

Looks good then, just seemed odd to me for some reason.

**Pull request closed without merging**